apps/mobile/src/features/activity/utils/format-activity.ts

```diff
@@ -1,52 +1,134 @@
+  status: BaseOnChainActivity['status'];
+}
+
+export function formatActivityStatus({ type, status }: FormatActivityStatusProps) {
+  switch (true) {
+    case type === 'sendAsset' && status === 'success':
+      return t({
+        id: 'activity.status.sent',
+        message: 'Sent',
+      });
+    case type === 'sendAsset' && status === 'failed':
+      return t({
+        id: 'activity.status.send-failed',
+        message: 'Send Failed',
+      });
+    case type === 'sendAsset' && status === 'pending':
+      return t({
+        id: 'activity.status.sending',
+        message: 'Sending',
+      });
+    case type === 'receiveAsset' && status === 'success':
+      return t({
+        id: 'activity.status.received',
+        message: 'Received',
+      });
+    case type === 'receiveAsset' && status === 'failed':
+      return t({
+        id: 'activity.status.receive-failed',
+        message: 'Receive fail',
+      });
+    case type === 'executeSmartContract' && status === 'success':
+      return t({
+        id: 'activity.status.executed',
+        message: 'Executed',
+      });
+    case type === 'executeSmartContract' && status === 'pending':
+      return t({
+        id: 'activity.status.executing',
+        message: 'Executing',
+      });
+    case type === 'executeSmartContract' && status === 'failed':
+      return t({
+        id: 'activity.status.execute-failed',
+        message: 'Execution failed',
+      });
+    case type === 'deploySmartContract' && status === 'success':
+      return t({
+        id: 'activity.status.deployed',
+        message: 'Deployed',
+      });
+    case type === 'deploySmartContract' && status === 'pending':
+      return t({
+        id: 'activity.status.deploying',
+        message: 'Deploying',
+      });
+    case type === 'deploySmartContract' && status === 'failed':
+      return t({
+        id: 'activity.status.deploy-failed',
+        message: 'Deployment failed',
+      });
+    default:
+      return undefined;
+  }
+}
+
+interface FormatActivityCaptionProps {
+  type: OnChainActivity['type'];
+  status: BaseOnChainActivity['status'];
+  timestamp: number;
+}
+
+export function formatActivityCaption({ type, status, timestamp }: FormatActivityCaptionProps) {
+  const isRecent = dayjs(timestamp).isAfter(dayjs().subtract(1, 'hour'));
+  const time = dayjs(timestamp * 1000).format('MMM D, YYYY');
+
+  const timestampText = isRecent
+    ? `${dayjs().diff(dayjs(timestamp * 1000), 'minute')} ${t({
+        id: 'activity.time.minutes-ago',
+        message: 'minutes ago',
+      })}`
+    : time;
+
+  const statusText = formatActivityStatus({ type, status });
+  return statusText ? `${statusText} ${timestampText}` : timestampText;
+}
+
+export function getActivityTitle(activity: OnChainActivity) {
+  switch (activity.type) {
+    case 'sendAsset':
+    case 'receiveAsset':
+      if (!activity.value?.crypto?.symbol) {
+        // TODO ENG-37 - find out about sBTC rewards - Asked design
```

Comment:
I'll do more work on this soon as we got some [new designs](https://linear.app/leather-io/issue/ENG-37/adjust-and-update-activity-flag-itens-state-according-to-designs#comment-c068893d) for this now